engine/gamesys/src/gamesys/test/test_gamesys_http.cpp

```diff
@@ -0,0 +1,124 @@
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include <script/test_script.h>
+#include <dlib/dstrings.h>
+#include <dlib/time.h>
+
+#include "test_gamesys.h"
+
+using namespace dmVMath;
+
+static int g_HttpPort = 9001;
+static char g_HttpAddress[128] = "localhost";
+
+#define DEFAULT_URL "__default_url"
+
+static void SetHttpAddress(lua_State* L)
+{
+    char buf[128];
+    dmSnPrintf(buf, sizeof(buf), "IP='%s'\n", g_HttpAddress);
+    dmScriptTest::RunString(L, buf);
+    dmSnPrintf(buf, sizeof(buf), "PORT=%d\n", g_HttpPort);
+    dmScriptTest::RunString(L, buf);
+    dmSnPrintf(buf, sizeof(buf), "ADDRESS='http://%s:%d'\n", g_HttpAddress, g_HttpPort);
+    dmScriptTest::RunString(L, buf);
+}
+
+TEST_F(ComponentTest, HTTPRequest)
+{
+    dmGameSystem::ScriptLibContext scriptlibcontext;
+    scriptlibcontext.m_Factory         = m_Factory;
+    scriptlibcontext.m_Register        = m_Register;
+    scriptlibcontext.m_LuaState        = dmScript::GetLuaState(m_ScriptContext);
+    scriptlibcontext.m_GraphicsContext = m_GraphicsContext;
+    scriptlibcontext.m_ScriptContext   = m_ScriptContext;
+
+    dmGameSystem::InitializeScriptLibs(scriptlibcontext);
+
+    lua_State* L = scriptlibcontext.m_LuaState;
+
+    dmMessage::URL m_DefaultURL;
+    ASSERT_EQ(dmMessage::RESULT_OK, dmMessage::NewSocket("default_socket", &m_DefaultURL.m_Socket));
+    m_DefaultURL.m_Path = dmHashString64("default_path");
+    m_DefaultURL.m_Fragment = dmHashString64("default_fragment");
+    dmScript::PushURL(scriptlibcontext.m_LuaState, m_DefaultURL);
+    lua_setglobal(L, DEFAULT_URL);
+
+    SetHttpAddress(L);
+
+    ASSERT_TRUE(dmGameObject::Init(m_Collection));
+
+    dmGameObject::HInstance go = Spawn(m_Factory, m_Collection, "/http/http_request.goc", dmHashString64("/http_request"), 0, 0, Point3(0, 0, 0), Quat(0, 0, 0, 1), Vector3(1, 1, 1));
+    ASSERT_NE((void*)0, go);
+
+    uint64_t stop_time = dmTime::GetTime() + 1*1e6; // 1 second
+    bool tests_done = false;
+    while (dmTime::GetTime() < stop_time && !tests_done)
+    {
+    	ASSERT_TRUE(dmGameObject::Update(m_Collection, &m_UpdateContext));
+
+        // check if tests are done
+        lua_getglobal(L, "tests_done");
+        tests_done = lua_toboolean(L, -1);
+        lua_pop(L, 1);
+
+        dmTime::Sleep(30*1000);
+    }
+
+    ASSERT_TRUE(tests_done);
+    ASSERT_TRUE(dmGameObject::Final(m_Collection));
+
+    dmGameSystem::FinalizeScriptLibs(scriptlibcontext);
+}
+
+int main(int argc, char **argv)
```

Comment:
This file is here for two reasons:
* I don't want to mess the current HTTP tests up that was moved into gamesys
* I don't want to put this into the main gamesys test because it relies on a http server running, and the gamesys test is getting a bit out of hand size wise